pkgs/shells/rc/default.nix

```diff
@@ -0,0 +1,48 @@
+{ stdenv, fetchurl, autoreconfHook
+, ncurses #acinclude.m4 wants headers for tgetent().
+, historySupport ? false
+, editSupport ? "readline", readline ? null, libedit ? null
+}:
+
+assert editSupport == "readline" -> readline != null;
```

Comment:
The configure script defaults on null regardless of availability so that's not an option. Otherwise:
1. _readline_ is 20k of buggy blubber. But it works for tab-completion.
2. I'm not getting tab-completion to work with the 5k loc _libedit_ so I'm forced to compromise on _readline_ for now.
3. Upstream is reviewing the 1.1k loc _linenoise_ as a line editing library substitute ( https://github.com/rakitzis/rc/pull/5 ).
4. Users who just want to execute rc scripts wouldn't need any line editing library.

Regarding the style, I just prefer passing a message over sharing a resource. Like, wouldn't a third party overriding rc will have to pull the line editing library it wants into it's own deceleration? e.g.:
```
buildInputs = [
  (rc.override {readline = null; libedit = libedit;})
];
```
But really, I'm fine either way so just link here an example for what you want and I'll push the changes.